api/app/Services/Zoom/ZoomService.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace App\Services\Zoom;
+
+use App\Entity\Event;
+use App\Repositories\User\UserRepositoryInterface;
+use Illuminate\Support\Facades\Http;
+
+final class ZoomService
+{
+    private const ZOOM_MEETING_API_ENDPOINT = 'https://api.zoom.us/v2/users/me/meetings';
+    private const ZOOM_GET_USER_API_ENDPOINT = 'https://api.zoom.us/v2/users/me';
+    private const ZOOM_OAUTH_TOKEN_POST_REQUEST_URL = 'https://zoom.us/oauth/token';
+
+    private UserRepositoryInterface $userRepository;
+
+    public function __construct(UserRepositoryInterface $userRepository)
+    {
+        $this->userRepository = $userRepository;
+    }
+
+    public function meeting(Event $event): string
+    {
+        $user = $this->userRepository->getById($event->eventType->owner_id);
+
+        $response = Http::withHeaders([
+            "Authorization" => "Bearer " . $user->zoom_access_token,
+        ])->post(self::ZOOM_MEETING_API_ENDPOINT, [
+            "topic" => $event->eventType->name,
+            "start_time" => $event->start_date,
+            "timezone" => $event->timezone
+        ]);
+
+        $zoomResponseArr = json_decode($response->body(), true);
+
+        return $zoomResponseArr['start_url'];
+    }
+
+    public function saveToken($code)
+    {
+        $token = $this->getZoomToken($code);
+        $zoomUser = $this->getZoomUser($token);
+
+        $user = $this->userRepository->getByEmail($zoomUser['email']);
```

Comment:
Use `state` query parameter (https://marketplace.zoom.us/docs/guides/auth/oauth) to pass the user ID instead of email